be/src/storage/rowset/segment_v2/column_reader.cpp

```diff
@@ -510,24 +514,39 @@ Status ColumnReader::new_iterator(ColumnIterator** iterator) {
     }
 }
 
-Status ColumnReader::ensure_index_loaded(ReaderType reader_type) {
-    Status status = _load_ordinal_index_once.call([this] {
+Status ColumnReader::_load_zone_map_index_once() {
+    Status status = _zonemap_index_once.call([this] {
+        RETURN_IF_ERROR(_load_zone_map_index(!config::disable_storage_page_cache, _opts.kept_in_memory));
+        return Status::OK();
+    });
+    return status;
+}
+
+Status ColumnReader::_load_bitmap_index_once() {
+    Status status = _bitmap_index_once.call([this] {
+        RETURN_IF_ERROR(_load_bitmap_index(!config::disable_storage_page_cache, _opts.kept_in_memory));
+        return Status::OK();
+    });
+    return status;
+}
+
+Status ColumnReader::_load_bloom_filter_index_once() {
+    Status status = _bloomfilter_index_once.call([this] {
+        RETURN_IF_ERROR(_load_bloom_filter_index(!config::disable_storage_page_cache, _opts.kept_in_memory));
+        return Status::OK();
+    });
+    return status;
+}
+
+
+Status ColumnReader::ensure_index_loaded() {
```

Comment:
I think this name is not suitable. You can change it to _load_ordinal_index

Reply:
I change it to load_ordinal_index_once()